app/src/main/java/com/example/androidacademyhw/movies/MoviesViewModel.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.androidacademyhw.movies
+
+import android.app.Application
+import android.util.Log
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.androidacademyhw.data.Movie
+import com.example.androidacademyhw.data.loadMovies
+import kotlinx.coroutines.*
+
+class MoviesViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val exceptionHandler = CoroutineExceptionHandler { canceledContext, exception ->
+        val isActive = coroutineScope.isActive
+        Log.d(
+            FragmentMoviesList.TAG,
+            "ExceptionHandler [Scope active:$isActive, canceledContext:$canceledContext]"
+        )
+        coroutineScope.launch {
+            logException(exception)
+        }
```

Comment:
i do not remember exactly, do u need the launch here for logging?

Reply:
Me neither :D
Any suggestions?

Reply:
try to delete then and see

Reply:
Seems like we need it. 
logExeption() is a suspend function